apps/src/sites/code.org/pages/public/teacher-dashboard/index.js

```diff
@@ -45,17 +45,39 @@ function renderSectionProjects(sectionId) {
 // TODO (bjvanminnen): Fix remaining lint errors and re-enable rules.
 /* eslint-disable eqeqeq, no-unused-vars */
 function main() {
-
-
+  const studioUrlPrefix = data.studiourlprefix;
   var valid_scripts = data.valid_scripts;
   var valid_courses = data.valid_courses;
-  var homepage_url = data.homepage_url;
   var hoc_assign_warning = data.hoc_assign_warning;
   var disabled_scripts = data.disabled_scripts;
   var i18n = data.i18n;
   var error_string_none_selected = i18n.error_string_none_selected;
   var error_string_other_section = i18n.error_string_other_section;
 
+  // Sections can be assigned to either a course or a script. Since there's a
+  // possibility that we could have a script and a course with the same id, we
+  // need a way to differentiate them. We do that by giving scripts and courses
+  // assignment ids, which are guaranteed to be unique across the population of
+  // both courses and scripts.
```

Comment:
I don't know that I love this solution, but it seems okay.

We could have the server provide us with assign_ids for these object instead of doing it on the client. This leads to (arguably) cleaner client code, but feels strange given that the server doesn't need this assignment_id for anything.